src/config_param.h

```diff
@@ -172,189 +172,189 @@ using DoubleConfigParam = RangeConfigParam<double>;
-    std::vector<T> _valid = {};
+		void Disable() {
+			_valid.clear();
+		}
+
+		void Lock(T value) {
+			_value = value;
+			_valid = { _value };
+		}
+	private:
+		T _value = {};
+		std::vector<T> _valid = {};
 };
 
 template <typename E>
 class EnumConfigParam {
-public:
-	using value_type = E;
-
-    explicit EnumConfigParam(E value = {}) : _value{ value } {}
-    EnumConfigParam(E value, std::initializer_list<E> valid) : _value{ value }, _valid{ valid } { _valid[_value] = true; }
-
-    bool Set(E value) {
-        if (IsValid(value)) {
-            _value = value;
-            return true;
-        }
-        return false;
-    }
-
-    const E& Get() const {
-		return _value;
-	}
-
-    bool IsValid(const E& value) const {
-		return _valid[value];
-	}
-
-	bool Enabled() const {
-		return _valid.any();
-	}
-
-	bool Locked() const {
-		return _valid.count() <= 1;
-	}
-
-    void ReplaceValidSet(lcf::FlagSet<E> valid) {
-		_valid = std::move(valid);
-		if (Enabled() && !IsValid(_value)) {
-			_value = GetFirstValid();
-		}
-	}
-
-	void AddToValidSet(E value) {
-		_valid[value] = true;
-	}
-
-	void RemoveFromValidSet(const E& value) {
-		_valid[value] = false;
-		if (Enabled() && !IsValid(_value)) {
-			_value = GetFirstValid();
-		}
-	}
-
-	void Disable() {
-		_valid = {};
-	}
-
-	void Lock(E value) {
-		_value = value;
-		_valid = {};
-		_valid[value] = true;
-	}
-private:
-    E _value = {};
-	lcf::FlagSet<E> _valid = ~lcf::FlagSet<E>();
-
-	E GetFirstValid() const {
-		for (size_t i = 0; i < _valid.size(); ++i) {
-			auto e = static_cast<E>(i);
-			if (_valid[e]) {
-				return e;
+	public:
+		using value_type = E;
+
+		explicit EnumConfigParam(E value = {}) : _value{ value } {}
+		EnumConfigParam(E value, std::initializer_list<E> valid) : _value{ value }, _valid{ valid } { _valid[_value] = true; }
+
+		bool Set(E value) {
+			if (IsValid(value)) {
+				_value = value;
+				return true;
+			}
+			return false;
+		}
+
+		const E& Get() const {
+			return _value;
+		}
+
+		bool IsValid(const E& value) const {
+			return _valid[value];
+		}
+
+		bool Enabled() const {
+			return _valid.any();
+		}
+
+		bool Locked() const {
+			return _valid.count() <= 1;
+		}
+
+		void ReplaceValidSet(lcf::FlagSet<E> valid) {
+			_valid = std::move(valid);
+			if (Enabled() && !IsValid(_value)) {
+				_value = GetFirstValid();
+			}
+		}
+
+		void AddToValidSet(E value) {
+			_valid[value] = true;
+		}
+
+		void RemoveFromValidSet(const E& value) {
+			_valid[value] = false;
+			if (Enabled() && !IsValid(_value)) {
+				_value = GetFirstValid();
+			}
+		}
+
+		void Disable() {
+			_valid = {};
+		}
+
+		void Lock(E value) {
+			_value = value;
+			_valid = {};
+			_valid[value] = true;
+		}
+	private:
+		E _value = {};
+		lcf::FlagSet<E> _valid = ~lcf::FlagSet<E>();
+
+		E GetFirstValid() const {
+			for (size_t i = 0; i < _valid.size(); ++i) {
+				auto e = static_cast<E>(i);
+				if (_valid[e]) {
+					return e;
+				}
 			}
+			return E{};
```

Comment:
the last commit added tab before public:/private:.  I think your editor is too smart with formating :)